<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>results</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">partition_heritability</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/j3gu/partition_heritability">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">results</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-07-23
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>partition_heritability/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200622code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200622)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200622code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200622)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomj3gupartitionheritabilitytree53fb47d22b2af087ce589e710d75059e8234afbdtargetblank53fb47da"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/j3gu/partition_heritability/tree/53fb47d22b2af087ce589e710d75059e8234afbd" target="_blank">53fb47d</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomj3gupartitionheritabilitytree53fb47d22b2af087ce589e710d75059e8234afbdtargetblank53fb47da" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .ipynb_checkpoints/

Untracked files:
    Untracked:  README.html
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/compare_with_other_eQTLs_analyiss.Rmd
    Untracked:  output/total_h2_for_each_gene_chr22.txt

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/partition_heritability/blob/53fb47d22b2af087ce589e710d75059e8234afbd/analysis/results.Rmd" target="_blank">53fb47d</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-07-23
</td>
<td>
wflow_publish(“analysis/results.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/partition_heritability/0d6c0964afdeb773cbf7f4970971fb839563fff4/docs/results.html" target="_blank">0d6c096</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-07-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/partition_heritability/blob/9249dba85fa148462cf766636b39e565615893dc/analysis/results.Rmd" target="_blank">9249dba</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-07-23
</td>
<td>
wflow_publish(“analysis/results.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/partition_heritability/8b988808c41214171cefeebfefbba6e4424df84e/docs/results.html" target="_blank">8b98880</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-07-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/partition_heritability/blob/549e1b38bdc722209587b1dba36f0b38dddaf175/analysis/results.Rmd" target="_blank">549e1b3</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-07-13
</td>
<td>
run analysis over chr21 and chr22
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/partition_heritability/6681a1f14453baedd3f2a84c9029eb980326c638/docs/results.html" target="_blank">6681a1f</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-07-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/partition_heritability/9df76c1b3fcba41d239c46cf2ee35af7a357f5fe/docs/results.html" target="_blank">9df76c1</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-07-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/partition_heritability/blob/b8ec4db63eadf2c0ef09b4cd5a5d9ac07e27e3a6/analysis/results.Rmd" target="_blank">b8ec4db</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-07-09
</td>
<td>
wflow_publish(“analysis/*“)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="scale-h2-for-each-gene" class="section level3">
<h3>Scale h2 for each gene</h3>
<ol style="list-style-type: decimal">
<li>The numbers of cis SNPs for each gene were calcualted. Cis SNPs are defined as +/- 1Mb away from TSS of genes.</li>
<li>The h2 value for each gene from LDSC output was scaled as <span class="math display">\[
    \hat h_{s}^2 = \hat h^2 * \frac{\text{# of cis SNPs}}{M_{total}}
\]</span> , where <span class="math inline">\(M_{total} = 9254535\)</span> - the total number of genome-wide SNPs from 1000 genome.</li>
</ol>
</div>
<div id="summarize-snp-heritabilities-for-gene-expression" class="section level3">
<h3>Summarize SNP heritabilities for gene expression</h3>
<ul>
<li>Any genes with negative total h2 were excluded in the analysis.</li>
</ul>
<pre class="r"><code>ref_dir&lt;-&quot;/project2/xuanyao/jing/data/eQTLGen&quot;
out_dir&lt;-&quot;/project2/xuanyao/jing/project/results/total_h2&quot;

h2&lt;-lapply(1:22, function(i){
  f&lt;-read.table(paste0(out_dir, sprintf(&quot;/chr%d_total_h2_stats.txt&quot;, i)), header = T)
  chr&lt;-rep(paste0(&quot;chr&quot;, i), dim(f)[1])
  return(data.frame(f, chr))
})
h2_df&lt;-do.call(rbind, h2)
print(paste0(&quot;The mean total h2 across genes without any filtering is &quot;, round(mean(h2_df$h2[h2_df$h2 &gt;= 0]),2)))</code></pre>
<pre><code>[1] &quot;The mean total h2 across genes without any filtering is 0.04&quot;</code></pre>
<pre class="r"><code>print(paste0(&quot;The standard deviation of total h2 values across genes are &quot;, sd(h2_df[h2_df$h2 &gt;= 0, &quot;sd&quot;])))</code></pre>
<pre><code>[1] &quot;The standard deviation of total h2 values across genes are 2566.80482162777&quot;</code></pre>
<pre class="r"><code>print(paste0(round(100*sum(h2_df$h2[h2_df$h2 &gt;=0]&gt;=1)/dim(h2_df)[1],1), &quot;% of the genes have estimated h2 to be higher than 1.&quot;))</code></pre>
<pre><code>[1] &quot;0.1% of the genes have estimated h2 to be higher than 1.&quot;</code></pre>
<ul>
<li>Any genes with estimated h2 higher than one were excluded in the analysis.<br />
Average heritability estimate was calculated as</li>
</ul>
<p><span class="math display">\[
    \frac{\sum_{1}^nh^2_{g}}{n}
\]</span></p>
<pre class="r"><code>h2_df_raw&lt;-h2_df
h2_df&lt;-h2_df[h2_df$h2 &gt;=0 &amp; h2_df$h2&lt;=1, ]
print(paste0(&quot;Average heritability estimate was calculated to be &quot;, round(sum(h2_df$h2)/dim(h2_df)[1],3)))</code></pre>
<pre><code>[1] &quot;Average heritability estimate was calculated to be 0.03&quot;</code></pre>
<pre class="r"><code>#take count of standard deviation
h2_df[,&quot;Z_score&quot;]&lt;-(h2_df$h2-mean(h2_df$h2))/h2_df$sd
hist(h2_df$Z_score, main = &quot;Transform estimated h2 into z scores&quot;)</code></pre>
<p><img src="figure/results.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/partition_heritability/blob/0d6c0964afdeb773cbf7f4970971fb839563fff4/docs/figure/results.Rmd/unnamed-chunk-2-1.png" target="_blank">0d6c096</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-07-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/partition_heritability/blob/8b988808c41214171cefeebfefbba6e4424df84e/docs/figure/results.Rmd/unnamed-chunk-2-1.png" target="_blank">8b98880</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-07-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#Further filter h2 by within two standard deviations of average h2
#h2_df_correct&lt;-h2_df
#h2_df&lt;-h2_df_correct[h2_df$Z_score&gt;=-2 &amp; h2_df$Z_score&lt;=2,] </code></pre>
<div id="observations" class="section level4">
<h4>Observations:</h4>
<ol style="list-style-type: decimal">
<li>The overall average cis-heritability estimate for blood was calculated to be 0.156, which is within the known range for cis-SNP heritability for RNASeq and ChIP-Seq (2%-20%).</li>
</ol>
<ul>
<li>Split by chromosome</li>
</ul>
<pre class="r"><code>avg_h2_by_chr&lt;-unlist(lapply(h2, function(i){
  return(round(sum(i$h2[i$h2 &gt;= 0])/dim(i)[1],3))
}))
avg_h2_df&lt;-data.frame(paste0(&quot;chr&quot;, 1:22), avg_h2_by_chr)
colnames(avg_h2_df)&lt;-c(&quot;chr&quot;, &quot;avg_h2&quot;)

p&lt;-ggplot(avg_h2_df, aes(x = chr, y = avg_h2)) + 
    geom_bar(stat = &quot;identity&quot;) + 
    theme(axis.text.x = element_text(angle = 45, vjust = -0.000005)) +
    geom_hline(yintercept=0.03, color = &quot;red&quot;, linetype=&quot;dashed&quot;)
p</code></pre>
<p><img src="figure/results.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/partition_heritability/blob/0d6c0964afdeb773cbf7f4970971fb839563fff4/docs/figure/results.Rmd/unnamed-chunk-3-1.png" target="_blank">0d6c096</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-07-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/partition_heritability/blob/8b988808c41214171cefeebfefbba6e4424df84e/docs/figure/results.Rmd/unnamed-chunk-3-1.png" target="_blank">8b98880</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-07-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>###Violin plots of distribution of h2 for each chromosome
p2&lt;-ggplot(h2_df_raw, aes(x = chr, y = h2)) + geom_violin() +
  theme(axis.text.x = element_text(angle = 45)) +
  ylab(&quot;h2 (All genes included)&quot;)
  ggtitle(&quot;Distribution of estimated h2 across chromosomes&quot;)</code></pre>
<pre><code>$title
[1] &quot;Distribution of estimated h2 across chromosomes&quot;

attr(,&quot;class&quot;)
[1] &quot;labels&quot;</code></pre>
<pre class="r"><code>p2</code></pre>
<p><img src="figure/results.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-2">
Past versions of unnamed-chunk-3-2.png
</button>
</p>
<div id="fig-unnamed-chunk-3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/partition_heritability/blob/0d6c0964afdeb773cbf7f4970971fb839563fff4/docs/figure/results.Rmd/unnamed-chunk-3-2.png" target="_blank">0d6c096</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-07-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/partition_heritability/blob/8b988808c41214171cefeebfefbba6e4424df84e/docs/figure/results.Rmd/unnamed-chunk-3-2.png" target="_blank">8b98880</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-07-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>p3&lt;-ggplot(h2_df, aes(x = chr, y = h2)) + geom_violin() +
  geom_boxplot(width=.1) +
  theme(axis.text.x = element_text(angle = 45)) +
  ylab(&quot;filtered_h2&quot;)
  ggtitle(&quot;Distribution of estimated h2 across chromosomes&quot;)</code></pre>
<pre><code>$title
[1] &quot;Distribution of estimated h2 across chromosomes&quot;

attr(,&quot;class&quot;)
[1] &quot;labels&quot;</code></pre>
<pre class="r"><code>p3</code></pre>
<p><img src="figure/results.Rmd/unnamed-chunk-3-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-3">
Past versions of unnamed-chunk-3-3.png
</button>
</p>
<div id="fig-unnamed-chunk-3-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/partition_heritability/blob/0d6c0964afdeb773cbf7f4970971fb839563fff4/docs/figure/results.Rmd/unnamed-chunk-3-3.png" target="_blank">0d6c096</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-07-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="observations-1" class="section level4">
<h4>Observations:</h4>
<ol style="list-style-type: decimal">
<li>The average heritability for gene expression varies across chromosomes. Chr7, chr14, chr16 and chr19 have higher average heritability compared to other chromosomes.<br />
</li>
<li>The distributions of estimated heritability differ between chromosomes. Certain chromosomes such as chr5, chr6, and chr14 have extreme negative h2, while others have h2 values higher than one.<br />
</li>
<li>After filtering the negative and greater than one h2 values, estimated h2 across chromosomes dislay simialr distributions.</li>
</ol>
</div>
<div id="correlation-between-cis-snps-heritatblities-with-gene-characteristics" class="section level4">
<h4>Correlation between cis-SNPs’ heritatblities with gene characteristics</h4>
<ol style="list-style-type: decimal">
<li><strong>gene length (overall)</strong></li>
</ol>
<pre class="r"><code>plot(x=h2_df$gene_length, y=h2_df$h2, xlab = &quot;gene length&quot;, ylab = &quot;estimated h2&quot;,
     main = &quot;Heritabiltiy vs.gene length&quot;)</code></pre>
<p><img src="figure/results.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/partition_heritability/blob/0d6c0964afdeb773cbf7f4970971fb839563fff4/docs/figure/results.Rmd/unnamed-chunk-4-1.png" target="_blank">0d6c096</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-07-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/partition_heritability/blob/8b988808c41214171cefeebfefbba6e4424df84e/docs/figure/results.Rmd/unnamed-chunk-4-1.png" target="_blank">8b98880</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-07-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/partition_heritability/blob/9df76c1b3fcba41d239c46cf2ee35af7a357f5fe/docs/figure/results.Rmd/unnamed-chunk-4-1.png" target="_blank">9df76c1</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-07-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cor(h2_df$gene_length, h2_df$h2)</code></pre>
<pre><code>[1] -0.04826811</code></pre>
<ol start="2" style="list-style-type: decimal">
<li><strong>gene length by binS</strong><br />
Note:<br />
</li>
</ol>
<ul>
<li>equal number of genes in each bin</li>
</ul>
<pre class="r"><code>h2_df[, &quot;bin&quot;]&lt;-cut_number(h2_df[,&quot;gene_length&quot;], n = 20)

ggplot(h2_df, aes(x=bin, y=h2) ) +
    geom_boxplot(fill=&quot;#69b3a2&quot;) +
    xlab(&quot;Gene length by bins&quot;) + 
    ylab(&quot;Estimated h2&quot;) + 
    theme(axis.text.x = element_text(angle = 90, vjust = -0.5)) +
    ggtitle(&quot;Estimated h2 against gene lengths divided into 20 bins&quot;)</code></pre>
<p><img src="figure/results.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/partition_heritability/blob/0d6c0964afdeb773cbf7f4970971fb839563fff4/docs/figure/results.Rmd/unnamed-chunk-5-1.png" target="_blank">0d6c096</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-07-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="observations-2" class="section level4">
<h4>Observations:</h4>
<ol style="list-style-type: decimal">
<li>Overall, estimated heratibility has no correlation with gene length based on correlation coefficient and scatter plot.<br />
</li>
<li><p>When we divided gene length into bins, genes in the lowest three or four quantiles seem to have higher estimated h2 values. More rigorous statistical tests need to be performed to compare the distributions. Nontheless, it is not obvious to me that there is a trend among the h2 with respect to gene length.</p></li>
<li><p><strong>Convervation scores</strong><br />
<img src="figure/results.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p></li>
</ol>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/partition_heritability/blob/0d6c0964afdeb773cbf7f4970971fb839563fff4/docs/figure/results.Rmd/unnamed-chunk-6-1.png" target="_blank">0d6c096</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-07-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/results.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-2">
Past versions of unnamed-chunk-6-2.png
</button>
</p>
<div id="fig-unnamed-chunk-6-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/partition_heritability/blob/0d6c0964afdeb773cbf7f4970971fb839563fff4/docs/figure/results.Rmd/unnamed-chunk-6-2.png" target="_blank">0d6c096</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-07-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="observations-3" class="section level4">
<h4>Observations:</h4>
<ol style="list-style-type: decimal">
<li>The scatter plot shows that genes are either highly conserved or not conversed are enriched with low estimated h2. Genes with low pLI scores seem have a higher proportion with high estimated h2.<br />
</li>
<li>Again, the trend is not obvious when dividing conservation scores by bins.</li>
</ol>
<p>4.<strong>EDS scores and number of enhancers</strong></p>
<pre class="r"><code>h2_df[,&quot;EDS&quot;]&lt;-unlist(lapply(h2_df$gene_id, function(i){
  if(as.character(i) %in% ref$GeneSymbol){
    return(ref[ref$GeneSymbol == as.character(i), &quot;EDS&quot;][1])
  }else{
    return(NA)
  }
}))

h2_df[, &quot;EDS_bin&quot;]&lt;-cut_number(h2_df[,&quot;EDS&quot;], n = 20)

plot(y = h2_df$h2, x = h2_df$EDS, main = &quot;Estimated heritability vs. EDS Scores&quot;,
     xlab = &quot;EDS scores&quot;, ylab = &quot;Estimated h2&quot;)</code></pre>
<p><img src="figure/results.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/partition_heritability/blob/0d6c0964afdeb773cbf7f4970971fb839563fff4/docs/figure/results.Rmd/unnamed-chunk-7-1.png" target="_blank">0d6c096</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-07-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(h2_df[!is.na(h2_df$EDS),], aes(x=EDS_bin, y=h2) ) +
    geom_boxplot(fill=&quot;#69b3a2&quot;) +
    xlab(&quot;EDS scores by bins&quot;) + 
    ylab(&quot;Estimated h2&quot;) + 
    theme(axis.text.x = element_text(angle = 90, vjust = -0.5)) +
    ggtitle(&quot;Estimated h2 against EDS scores divided into 20 bins&quot;)</code></pre>
<p><img src="figure/results.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-2">
Past versions of unnamed-chunk-7-2.png
</button>
</p>
<div id="fig-unnamed-chunk-7-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/partition_heritability/blob/0d6c0964afdeb773cbf7f4970971fb839563fff4/docs/figure/results.Rmd/unnamed-chunk-7-2.png" target="_blank">0d6c096</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-07-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>h2_df[,&quot;num_enhancers&quot;]&lt;-unlist(lapply(h2_df$gene_id, function(i){
  if(as.character(i) %in% ref$GeneSymbol){
    return(ref[ref$GeneSymbol == as.character(i), &quot;ActivityLinking_EnhancerNumber&quot;][1])
  }else{
    return(NA)
  }
}))

plot(y = h2_df$h2, x = h2_df$num_enhancers, main = &quot;Estimated heritability vs. Activity_linking_enhancer_numbers&quot;,
     xlab = &quot;Enhancer numbers&quot;, ylab = &quot;Estimated h2&quot;)</code></pre>
<p><img src="figure/results.Rmd/unnamed-chunk-7-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-3">
Past versions of unnamed-chunk-7-3.png
</button>
</p>
<div id="fig-unnamed-chunk-7-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/partition_heritability/blob/0d6c0964afdeb773cbf7f4970971fb839563fff4/docs/figure/results.Rmd/unnamed-chunk-7-3.png" target="_blank">0d6c096</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-07-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>h2_df[, &quot;num_enhancers_bin&quot;]&lt;-cut_number(h2_df[,&quot;num_enhancers&quot;], n = 20)

ggplot(h2_df[!is.na(h2_df$num_enhancers),], aes(x=num_enhancers_bin, y=h2) ) +
    geom_boxplot(fill=&quot;#69b3a2&quot;) +
    xlab(&quot;Number of enhancers by bins&quot;) + 
    ylab(&quot;Estimated h2&quot;) + 
    theme(axis.text.x = element_text(angle = 90, vjust = -0.5)) +
    ggtitle(&quot;Estimated h2 against activity linking enhancer numbers divided into 20 bins&quot;)</code></pre>
<p><img src="figure/results.Rmd/unnamed-chunk-7-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-4">
Past versions of unnamed-chunk-7-4.png
</button>
</p>
<div id="fig-unnamed-chunk-7-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/partition_heritability/blob/0d6c0964afdeb773cbf7f4970971fb839563fff4/docs/figure/results.Rmd/unnamed-chunk-7-4.png" target="_blank">0d6c096</a>
</td>
<td>
Jing Gu
</td>
<td>
2020-07-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="observations-4" class="section level4">
<h4>Observations:</h4>
<ol style="list-style-type: decimal">
<li>The scatter plots of EDS scores and Enhancer numbers seem to be noisy to see the signal.<br />
</li>
<li>It is also hard to see the trend.</li>
</ol>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggplot2_3.1.1

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.4.6     compiler_3.5.1   pillar_1.3.1     later_0.7.5     
 [5] git2r_0.26.1     plyr_1.8.4       workflowr_1.6.0  tools_3.5.1     
 [9] digest_0.6.18    evaluate_0.12    tibble_2.1.1     gtable_0.2.0    
[13] pkgconfig_2.0.2  rlang_0.4.0      yaml_2.2.0       withr_2.1.2     
[17] stringr_1.3.1    dplyr_0.8.0.1    knitr_1.20       fs_1.3.1        
[21] rprojroot_1.3-2  grid_3.5.1       tidyselect_0.2.5 glue_1.3.0      
[25] R6_2.3.0         rmarkdown_1.10   purrr_0.3.2      magrittr_1.5    
[29] whisker_0.3-2    backports_1.1.2  scales_1.0.0     promises_1.0.1  
[33] htmltools_0.3.6  assertthat_0.2.0 colorspace_1.3-2 httpuv_1.4.5    
[37] labeling_0.3     stringi_1.2.4    lazyeval_0.2.1   munsell_0.5.0   
[41] crayon_1.3.4    </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
